### PR TITLE
Create Jetpack Redirect package

### PR DESCRIPTION
**Pull request by leogermani**
Creates a new package called Jetpack Redirect to handle the creation of redirect URLs.

This package is going to be once we merge #15140.

#### Testing instructions:

Nothing to test here. This PR only adds the package but it's not used anywhere. 

Just load the page and check if the site is still loading with no errors.

#### Proposed changelog entry for your changes:
* Added Jetpack Redirect package

### CODE REVIEW

File: packages/redirect/src/class-redirect.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	 *    @type string $path Additional path to be appended to the URL.
+	 *    @type string $query Query parameters to be added to the URL.
+	 *    @type string $anchor Anchor to be added to the URL.
+	 * }
+	 *
+	 * @return string The built URL
+	 */
+	public static function get_url( $source, $args = array() ) {
+
+		$url           = 'https://jetpack.com/redirect';
```
Comment (dereksmart) [inline packages/redirect/src/class-redirect.php:54]: I had a thought where we might want to make this filterable, but can't really think of any use-cases where someone would want to mimic this on their own server redirect. We can make it filterable later if desired.

**Review by dereksmart — DISMISSED** [review]

File: packages/redirect/composer.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+	"name": "automattic/jetpack-redirect",
+	"description": "Utilities to build URLs to the jetpack.com/redirect service",
+	"type": "library",
+	"license": "GPL-2.0-or-later",
+	"require": {
+		"automattic/jetpack-constants": "@dev"
```
Comment (dereksmart) [inline packages/redirect/composer.json:7]: Do we need this?

File: packages/redirect/src/class-redirect.php (hunk 1 above)
Comment (leogermani) [reply packages/redirect/src/class-redirect.php:54]: wouldn't it be a security riks? A plugin that hijacks all links from jetpack...?

File: packages/redirect/composer.json (hunk 2 above)
Comment (leogermani) [reply packages/redirect/composer.json]: No. https://github.com/Automattic/jetpack/pull/15434/commits/82c716d547818b75ca0c9dcf612fe06419dafb45

File: packages/redirect/src/class-redirect.php (hunk 1 above)
Comment (dereksmart) [reply packages/redirect/src/class-redirect.php:54]: A plugin could do a lot worse than that ;)

**Review by dereksmart — APPROVED** [review]

**Pull request merged**